src/ert/scheduler/job.py

```diff
@@ -147,30 +148,31 @@ async def run(
         checksum_lock: asyncio.Lock,
         max_submit: int = 1,
     ) -> None:
-        self._requested_max_submit = max_submit
-        for attempt in range(max_submit):
-            await self._submit_and_run_once(sem)
-
-            if self.returncode.cancelled() or self._scheduler._cancelled:
-                break
-
-            if self.returncode.result() == 0:
-                if self._scheduler._manifest_queue is not None:
-                    await self._verify_checksum(checksum_lock)
-                async with forward_model_ok_lock:
-                    await self._handle_finished_forward_model()
-                break
-
-            if attempt < max_submit - 1:
-                message = (
-                    f"Realization {self.iens} failed, "
-                    f"resubmitting for attempt {attempt+2} of {max_submit}"
-                )
-                logger.warning(message)
-                self.returncode = asyncio.Future()
-                self.started.clear()
-            else:
-                await self._send(JobState.FAILED)
+        with tracer.start_as_current_span(f"{__name__}.run.realization_{self.iens}"):
```

Comment:
I guess we might be a little lucky here since the scope will cover most, if not all, of the functions in this class. 👍